### DISCUSSION TITLE
No images and css in pdf with headless chromium

### DISCUSSION THREAD

**Discussion by dpfahlbusch** [pimcore/pimcore#13667] [Q&A / Support]
Hey fellow Pimcorenauts,
currently we have the problem that headless chromium won't render images and style.css files.

Had anyone the same problems and know how to solve it?

Cheers :)

**Comment by kingjia90** [comment]
Are you on Pimcore > 10.4.6? If yes, does the test print display the background as intended [here](https://github.com/pimcore/pimcore/pull/12691)
Just to make sure the configuration are fine, it just breaks "silently" if the values are not properly set.

Niklas had similar problems [here](https://github.com/pimcore/pimcore/issues/12666) and definitely fixed later on but i didn't got any details on what happened/caused that.

**Reply by NiklasBr** [reply]
It appears to work better if assets like images and style sheets are referenced with absolute URL (including domain).

We resolved the background issue with these CSS lines:

https://github.com/pimcore/pimcore/blob/b2ecc3ed925932ab62a3219e2853bb4dc577a54a/bundles/AdminBundle/templates/admin/settings/test_web2print.html.twig#L10-L11

But we are still having some CSS issues with Chromium (see #13493) @kingjia90 .

**Reply by kingjia90** [reply]
Thank you for sharing, this might help.

About the CSS Issues, i'd suggest to open a separate issue, because i find it would be "harder" to keep track and manage something mentioned in a comment under a PR after a long discussion, at least from planning and community feedback point of view

**Comment by passioneight** [comment]
@dpfahlbusch I had a similar issue, when generating PDFs locally. The issue, for me, lies somewhere in the background execution and I did not find the actual cause yet. So, as a quick-fix I set `disableBackgroundExecution`  to `true` (see `Processor` class) and used absolute URLs for assets.

Maybe this helps :-)

**Comment by BlackbitDevs** [comment]
When working with wkhtmltopdf we have used `<base href="https://www.example.org/">` and then it worked also with relative paths. But have not checked if this is the same with Headless Chromium.
